src/protocol/mqtt.rs

```diff
@@ -0,0 +1,123 @@
+use crate::extend::ExtendableThing;
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, skip_serializing_none, OneOrMany};
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub enum ControlPacket {
+    #[serde(rename = "mqv:publish")]
+    Publish,
+    #[serde(rename = "mqv:subscribe")]
+    Subscribe,
+    #[serde(rename = "mqv:unsubscribe")]
+    Unsubscribe,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub enum QoS {
+    #[serde(rename = "quality:0")]
+    AtMostOnce,
+    #[serde(rename = "quality:1")]
+    AtLeastOnce,
+    #[serde(rename = "quality:2")]
+    ExactlyOnce,
+}
+
+#[serde_as]
+#[skip_serializing_none]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash, Default)]
+pub struct Form {
+    #[serde(
+        rename = "mqv:retain",
+        skip_serializing_if = "std::ops::Not::not",
+        default
+    )]
+    pub retain: bool,
```

Comment:
This is more than fine -- in the rest of the Thing I am using `Option<T>` because I would like to have a proper type to avoid this kind of boilerplate (for both `bool` and container types). In this case I am happy to have the proper set of attributes :blush: .